modal-go/dict.go

```diff
@@ -0,0 +1,381 @@
+}
+
+// DictEphemeral creates a nameless, temporary Dict. Caller must CloseEphemeral.
+func DictEphemeral(ctx context.Context, options *EphemeralOptions) (*Dict, error) {
+	if options == nil {
+		options = &EphemeralOptions{}
+	}
+	var err error
+	ctx, err = clientContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.DictGetOrCreate(ctx, pb.DictGetOrCreateRequest_builder{
+		ObjectCreationType: pb.ObjectCreationType_OBJECT_CREATION_TYPE_EPHEMERAL,
+		EnvironmentName:    environmentName(options.Environment),
+	}.Build())
+	if err != nil {
+		return nil, err
+	}
+
+	heartbeatCtx, cancel := context.WithCancel(ctx)
+	d := &Dict{DictId: resp.GetDictId(), cancel: cancel, ephemeral: true, ctx: ctx}
+
+	// background heartbeat goroutine
+	go func() {
+		t := time.NewTicker(ephemeralObjectHeartbeatSleep)
+		defer t.Stop()
+		for {
+			select {
+			case <-heartbeatCtx.Done():
+				return
+			case <-t.C:
+				_, _ = client.DictHeartbeat(heartbeatCtx, pb.DictHeartbeatRequest_builder{
+					DictId: d.DictId,
+				}.Build()) // ignore errors – next call will retry or context will cancel
+			}
+		}
+	}()
+
+	return d, nil
+}
+
+// CloseEphemeral deletes an ephemeral Dict, only used with DictEphemeral.
+func (d *Dict) CloseEphemeral() {
+	if d.ephemeral {
+		d.cancel() // will stop heartbeat
+	} else {
+		// We panic in this case because of invalid usage. In general, methods
+		// used with `defer` like CloseEphemeral should not return errors.
+		panic(fmt.Sprintf("Dict %s is not ephemeral", d.DictId))
+	}
+}
+
+// DictLookup returns a handle to a (possibly new) Dict by deployment name.
+func DictLookup(ctx context.Context, name string, options *LookupOptions) (*Dict, error) {
+	if options == nil {
+		options = &LookupOptions{}
+	}
+	var err error
+	ctx, err = clientContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	creationType := pb.ObjectCreationType_OBJECT_CREATION_TYPE_UNSPECIFIED
+	if options.CreateIfMissing {
+		creationType = pb.ObjectCreationType_OBJECT_CREATION_TYPE_CREATE_IF_MISSING
+	}
+
+	resp, err := client.DictGetOrCreate(ctx, pb.DictGetOrCreateRequest_builder{
+		DeploymentName:     name,
+		EnvironmentName:    environmentName(options.Environment),
+		ObjectCreationType: creationType,
+	}.Build())
+	if err != nil {
+		return nil, err
+	}
+	return &Dict{ctx: ctx, DictId: resp.GetDictId()}, nil
+}
+
+// DictDelete removes a Dict by name.
+func DictDelete(ctx context.Context, name string, options *DeleteOptions) error {
+	if options == nil {
+		options = &DeleteOptions{}
+	}
+	var err error
+	ctx, err = clientContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	d, err := DictLookup(ctx, name, &LookupOptions{Environment: options.Environment})
+	if err != nil {
+		return err
+	}
+	_, err = client.DictDelete(ctx, pb.DictDeleteRequest_builder{DictId: d.DictId}.Build())
+	return err
+}
+
+// Get retrieves a value by key, with optional default value.
+func (d *Dict) Get(key any, defaultValue *any) (any, error) {
```

Comment:
One way to only support bytes + strings in go is to use generics: https://go.dev/doc/tutorial/generics#add_generic_function